### PR TITLE
Update developer guide for SelectCommand

### PR DESCRIPTION
**Pull request by TZL0**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+### Select Feature
+
+#### Current Implementation
+
+The select commands simulates a click on the 'PatientCard' or 'AppointmentCard' in the UI.
+
+The select methods are separated for patients and appointments, with command word 'selectpatient'
+and 'selectappointment' respectively. 
+
+The select commands makes use of the index of a patient or an appointment in the 'FilteredList's
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:289]: make use*

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+to identify whose information to show.
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:290]: which information

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+
+The 'SelectPatientCommandParser' and 'SelectAppointmentCommandParser' converts
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:292]: convert*

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+input String containing target index to the SelectCommand objects.
+
+On execution, the SelectPatientCommand will invoke the selectPatient() and selectAppointment() in the Model to
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:295]: update the FilteredPatientList as well

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+update the FilteredAppointmentList and FilteredBillList to contain selected patient's information only.
+
+Given below is an example usage scenario and how the find mechanism behaves at each step.
+
+Step 1. The user launches the application. All patients, appointments and bills are shown on different sections
+of the application as indexed lists.
+
+Step 2. The user executes `selectpatient 1` command to show all appointments and bills
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:303]: patient info also

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+
+Step 2. The user executes `selectpatient 1` command to show all appointments and bills 
+corresponding to the first listed patient.
+The `SelectPatient` command calls `Model#selectPatient(index)` to update the list of appointments
+and bills in the application.
+
+Step 3. The application displays the list of appointments and bills with the name equals to
+the first patient on the patient list panel.
+
+The select feature is now seperated for the patients, appointments and bills sections.
```
Comment (guokweijie) [inline docs/DeveloperGuide.md]: Our feature is not combined, as selectpatientcommand selects patient, appointments and bills tied to the patient.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+
+Design considerations:
+1. Length of command word
+2. Whether to use a prefix for the search term
+3. Number of keywords used for the search term
+
+Alternatives:
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:316]: can add more alternatives

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -277,6 +277,50 @@ Alternatives:
...
+
+1. Use a shorter command word (e.g. slp instead of selectpatient, sla instead of selectappointment)
+    - Pros: Easy to type
+    - Cons: May be confused with the select command for patients and appointments
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:320]: maybe u can change this cons to 'easier to type the wrong short-form command as they differ by 1 letter'

**Review by guokweijie — COMMENTED** [review]
just some minor fixes required

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (TZL0) [reply docs/DeveloperGuide.md:295]: No update to FilteredPatientList

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (TZL0) [reply docs/DeveloperGuide.md:303]: Patient info is always shown

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -277,6 +277,48 @@ Alternatives:
...
+the first patient on the patient list panel.
+
+The select feature is now seperated for the patients and appointments sections.
+
+Design considerations:
+1. Length of command word
+
+Alternatives:
+
+1. Use a shorter command word (e.g. slp instead of selectpatient, sla instead of selectappointment)
```
Comment (Fuzanwenn) [inline docs/DeveloperGuide.md:318]: can add more alternatives

**Review by Fuzanwenn — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (guokweijie) [reply docs/DeveloperGuide.md:303]: I mean the specific patient that we select

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (guokweijie) [reply docs/DeveloperGuide.md:295]: What do you mean update to?

**Review by guokweijie — APPROVED** [review]
LGTM

**Pull request merged**